src/main/java/com/iota/iri/network/HeartbeatPulseImpl.java

```diff
@@ -0,0 +1,62 @@
+package com.iota.iri.network;
+
+import com.iota.iri.network.neighbor.Neighbor;
+import com.iota.iri.network.protocol.Heartbeat;
+import com.iota.iri.service.snapshot.SnapshotProvider;
+import com.iota.iri.utils.thread.DedicatedScheduledExecutorService;
+import com.iota.iri.utils.thread.SilentScheduledExecutorService;
+
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Implementation of {@link HeartbeatPulse} interface.
+ */
+public class HeartbeatPulseImpl implements HeartbeatPulse {
+
+    private NeighborRouter neighborRouter;
+    private SnapshotProvider snapshotProvider;
+
+    /**
+     * The rate in milliseconds, at which heartbeats are sent out.
+     */
+    private static final int HEARTBEAT_RATE_MILLIS = 60000;
+
+    /**
+     * Holds a reference to the manager of the background worker.
+     */
+    private final SilentScheduledExecutorService executorService = new DedicatedScheduledExecutorService(
+            "Heartbeat Pulse");
+
+    public HeartbeatPulseImpl(NeighborRouter neighborRouter, SnapshotProvider snapshotProvider) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Public method and constructor comments are required](https://app.codacy.com/manual/GalRogozinski/iri/pullRequest?prid=5324203)